### ISSUE TITLE
🐞 [BUG] - newsletter form doesn't have proper label

### ISSUE THREAD

**Issue by 0xabdulkhalid** [anmode/grabtern-frontend#605]
### Description

- The newsletter form where the user can enter their Email addresses for Subscription is lacking proper labels.

### Current behaviour

- Due to the lacking label, accessibility devices can't properly announce the form's fields and purpose to the end user.

### Expected behaviour

- We must want to add proper label element associated with the input and also we want to add screen-reader only styles so that the label is hidden visually but for accessibility devices it's always visible.

### Reproduction steps

```bash
1. Go to 'https://www.grabtern.com/'
2. Right click on 'Email Address' input field on bottom of the page
3. See the lacking label for input element
```

### Screenshots

![Screenshot_2023-07-26-19-57-09_1366x768](https://github.com/anmode/grabtern-frontend/assets/92252895/1cf607be-8207-4786-bcfa-7a3bc6bbdc1d)

### Browsers

Firefox

### OS

Linux

### Are you contributing for a program ?

GSSoC2023

### Code of Conduct

- [X] I have read and agree to the Contributing Guidelines of this project.

**Comment by 0xabdulkhalid** [comment]
Hey there @anmode 👋🏻

Please make sure to assign this issue to me to make the work done.

Thank you!
